warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExpandMultiNormalizedTerms.java

```diff
@@ -235,71 +284,102 @@ protected JexlNode expandNodeForNormalizers(JexlNode node, Object data) {
+                                    }
+                                    failedNormalization = true;
                                 }
+                            } else {
+                                failedNormalization = true;
                             }
-                            // this could be CIDR notation, attempt to expand the node to the cidr range
                         } catch (Exception ne) {
                             if (log.isTraceEnabled()) {
                                 log.trace("Could not normalize " + term + " using " + normalizer.getClass());
                             }
-                            // if this was a regex, then lets assume that the regex could not be decoded enough to determine how to match against the index.
-                            // in this case we need to force this term to be evaluation only (i.e. cannot be done against an index)
-                            if (node instanceof ASTNRNode || node instanceof ASTERNode) {
-                                log.info("Pushing regex down to be evaluation only as it could not be normalized");
-                                evaluationOnlyRegex = true;
-                                // make sure we include the original form of the regex for this
+                            failedNormalization = true;
+                        }
+                    }
+                    
+                    // determine if we are marking this term as dropped or evaluation only
+                    boolean droppedExpression = false;
+                    boolean evaluationOnly = false;
+                    if (failedNormalization) {
+                        // if we are not being lenient then add the original term into the mix and make it eval only
+                        if (!lenient) {
+                            if (!normalizedTerms.contains(term)) {
+                                normalizedTerms.add(term);
+                                normalizedNodes.add(JexlNodeFactory.buildUntypedNode(node, fieldName, term));
+                            }
+                            evaluationOnly = true;
+                        }
+                        // else if we are being lenient and we have no successful normalizations, then drop the original term
+                        else if (normalizedNodes.isEmpty()) {
+                            if (!normalizedTerms.contains(term)) {
                                 normalizedTerms.add(term);
+                                normalizedNodes.add(JexlNodeFactory.buildUntypedNode(node, fieldName, term));
                             }
+                            droppedExpression = true;
                         }
                     }
                     
-                    if (normalizedTerms.size() > 1) {
+                    // build an expression from the normalized terms
+                    if (normalizedNodes.size() > 1) {
```

Comment:
My preference when creating a node from a variable length list is to do a switch statement on `normalizedNodes.size()` to get a clean logical progression like so
```
case 0:
case 1:
default:
```

Reply:
Just adding my 2 cents...  I think it's fine as-is.

Reply:
If it reads better for you, no problem switching it around.